### PR TITLE
ALFREDAPI-416 Removed max repository version from module.properties for builds for most recent Alfresco 

### PR DESCRIPTION
**Pull request by kpiot123**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-416

- [x] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [x] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [x] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation)?
- [x] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [x] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

**Comment by kpiot123** [comment]
Does this change also require an update of the changelog? Functionally nothing really changed.

**Comment by koenvangeerteruy** [comment]
@kpiot123 Changelog update? I think this change involves a change that can be of importance for end users.

**Comment by kpiot123** [comment]
> @kpiot123 Changelog update? I think this change involves a change that can be of importance for end users.

I wasn't sure but it can't hurt to add it. I have updated it.

File: apix-impl/52/src/main/config/module.properties [hunk 1]
```diff
@@ -3,5 +3,5 @@ module.title=${project.name}
 module.description=${project.description}
 module.version=${project.version}
 
-module.repo.version.min=5.2
-module.repo.version.max=5.2.99
+module.repo.version.min=${project.alfresco_min_version}
+module.repo.version.max=${project.properties["alfresco_max_version"]}
```
Comment (WimCrols) [inline apix-impl/52/src/main/config/module.properties:7]: Why not `${project.properties.alfresco_max_version}`?

File: apix-impl/build.gradle [hunk 2]
```diff
@@ -4,6 +4,31 @@ plugins {
...
+
+    if (currentProject.getName().equals(subProjectList.last().getName())) {
+        return ""
+    }
+
+    return minAlfrescoVersion + ".99"
+}
+
+ext {
+    alfresco_50_min_version = "5.0"
```
Comment (WimCrols) [inline apix-impl/build.gradle:25]: The versioning is already spread out over 5 files, so I prefer not to add another one. Can you remove these constants and simply hardcode the versions in the respective overrides.gradle?

Comment (koenvangeerteruy) [reply apix-impl/build.gradle:25]: OK. In that case we can also just remove or set the max_version in overrides.gradle to what it needs to be.
Unless, you want setting the max version to be automated when support for a new Alfresco version is added

Comment (kpiot123) [reply apix-impl/build.gradle:25]: To be clear, can the getMaxRepositoryVersionRequirement method be kept or should I also hard-code the max Alfresco version?

Comment (WimCrols) [reply apix-impl/build.gradle]: That's also ok for me.

Comment (WimCrols) [reply apix-impl/build.gradle]: To answer Koen: no, no automation is needed for that.

Comment (koenvangeerteruy) [reply apix-impl/build.gradle]: @WimCrols OK. So we set (hard-code) min and max Alfresco version in the override.gradle files?

Comment (WimCrols) [reply apix-impl/build.gradle]: Yes

File: apix-impl/61/overrides.gradle [hunk 3]
```diff
@@ -3,4 +3,5 @@ description = "Xenit API-X implementation Alfresco 6.1"
 ext {
     alfresco_version = alfresco_61_version
     alfresco_dm_version = alfresco_61_dm_version
+    alfresco_min_version = "6.1"
```
Comment (WimCrols) [inline apix-impl/61/overrides.gradle:6]: @kpiot123 could you add the comment here `Not setting max version to make it easier to test on next Alfresco version (only for last version)`, please? I will already approve. No need to assign back to me. After adding this, you may merge.

**Pull request merged**